src/app/clusters/time-synchronization-server/time-synchronization-server.h

```diff
@@ -0,0 +1,170 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file API declarations for time sync cluster.
+ */
+
+#pragma once
+
+#include "TimeSyncDataProvider.h"
+
+#include <app/server/Server.h>
+#include <app/util/af-types.h>
+#include <app/util/config.h>
+#include <lib/core/TLV.h>
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace TimeSynchronization {
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::TimeSynchronization;
+using namespace chip::app::Clusters::TimeSynchronization::Attributes;
+using chip::TimeSyncDataProvider;
+using chip::Protocols::InteractionModel::Status;
+
+class TimeSynchronizationServer
+{
+    struct timeZoneName
+    {
+        char name[64];
+    };
+
+public:
+    void Init();
+
+    static TimeSynchronizationServer & Instance(void);
+
+    TimeSyncDataProvider GetDataProvider(void) { return mTimeSyncDataProvider; }
+
+    CHIP_ERROR SetTrustedTimeSource(DataModel::Nullable<TimeSynchronization::Structs::TrustedTimeSourceStruct::Type> tts)
+    {
+        if (!tts.IsNull())
+        {
+            mTrustedTimeSource.SetNonNull(tts.Value());
```

Comment:
This and the corresponding bit in the other branch could both be replaced with `mTrustedTimeSource = tts`, no?